frontend/javascripts/dashboard/folders/metadata_table.tsx

```diff
@@ -0,0 +1,414 @@
+import {
+  DeleteOutlined,
+  FieldNumberOutlined,
+  FieldStringOutlined,
+  PlusOutlined,
+  TagsOutlined,
+} from "@ant-design/icons";
+import {
+  MenuProps,
+  InputNumberProps,
+  InputNumber,
+  Input,
+  Select,
+  Typography,
+  Dropdown,
+  Button,
+} from "antd";
+import { usePreviousValue } from "beautiful-react-hooks";
+import {
+  DatasetCollectionContextValue,
+  useDatasetCollectionContext,
+} from "dashboard/dataset/dataset_collection_context";
+import Toast from "libs/toast";
+import _ from "lodash";
+import React, { useEffect } from "react";
+import { useState } from "react";
+import { APIDataset, Folder, APIMetadata, APIMetadataType } from "types/api_flow_types";
+
+function getMetadataTypeLabel(type: APIMetadata["type"]) {
+  switch (type) {
+    case "string":
+      return (
+        <span>
+          <FieldStringOutlined /> Text
+        </span>
+      );
+    case "number":
+      return (
+        <span>
+          <FieldNumberOutlined /> Number
+        </span>
+      );
+    case "string[]":
+      return (
+        <span>
+          <TagsOutlined /> Multi-Item Text
+        </span>
+      );
+  }
+}
+
+const saveCurrentMetadata = async (
+  datasetOrFolderToUpdate: APIDataset | Folder,
+  metadata: IndexedMetadataEntries,
+  context: DatasetCollectionContextValue,
+  { dontUpdateState }: EffectCancelSignal = { dontUpdateState: false }, // equals whether the component is unmounted
```

Comment:
I'd rename dontUpdateState to `isComponentMounted` and add the comment that this is necessary to avoid the set-functions if it's not.
another idea: you could even wrap the set functions in the other component and adapt them so that they don't do anything if its not mounted. I think that would be cool because the wrapping would only happen once. currently, you need to define and update this signal once per useEffect definition.